field-agent/src/components/FieldAgents.js

```diff
@@ -0,0 +1,420 @@
+            dob,
+            heightInInches,
+            aliases: [],
+            agencies: []
+        };
+
+
+        setAgents([...agents, newAgent]);
+
+        resetForm();
+
+        setCurrentView('List');
+
+    }
+
+    const editAgentFormSubmitHandler = (event) => {
+        event.preventDefault();
+
+        const newAgent = {
+            id: editAgentId,
+            firstName,
+            middleName,
+            lastName,
+            dob,
+            heightInInches,
+            aliases: [],
+            agencies: []
+        };
+
+        const newAgents = [...agents];
+
+        const newAgentIndex = newAgents.findIndex(agent => agent.agentId === editAgentId);
+
+        newAgents[newAgentIndex] = newAgent;
+
+        setAgents(newAgents);
+
+        resetForm();
+
+        setCurrentView('List')
+    }
+
+    const agentEditClickHandler = (agentId) => {
+        setEditAgentId(agentId);
+
+        const editAgent = agents.find(agent => agent.agentId === agentId);
+
+        setFirstName(editAgent.firstName);
+        setMiddleName(editAgent.middleName);
+        setLastName(editAgent.lastName);
+        setDob(editAgent.dob);
+        setHeightInInches(editAgent.heightInInches);
+
+        setCurrentView('Edit');
+    }
+
+    const deleteAgentFormSubmitHandler = () => {
+        setAgents(agents.filter(agent => agent.agentId !== deleteAgentId));
+
+        setCurrentView('List');
+    }
+
+    const agentDeleteClickHandler = (agentId) => {
+
+        setDeleteAgentId(agentId);
+
+        const deleteAgent = agents.find(agent => agent.agentId === agentId);
+
+        setDeleteAgentId(deleteAgent.agentId)
+        setFirstName(deleteAgent.firstName);
+        setMiddleName(deleteAgent.middleName);
+        setLastName(deleteAgent.lastName);
+        setDob(deleteAgent.dob);
+        setHeightInInches(deleteAgent.heightInInches);
+
+        setCurrentView('Delete');
+
+    
+    };
+
+    const addAgentClickHandler = () => { setCurrentView('Add') }
+
+    const agentListClickHandler = () => { setCurrentView('List'); }
+
+    const homeClickHandler = () => { setCurrentView('Home') }
+
+    return (
+
+        <div>
+            {currentView === 'Add' ? (
+                <div class="container">
+                    <header>
+                        <h1>Field Agent</h1>
+                        <div>
+                            Welcome username! | <a href="#">Logout</a> | <a href="login.html">Login</a> | <a
+                                href="register.html">Register</a>
+                        </div>
+                    </header>
+                    <nav>
+                        <div>
+                            <button type="button" onClick={homeClickHandler} >Home</button> | <a href="#">Agencies</a> | <a href="#">Locations</a> |
+                            <button type="button" onClick={agentListClickHandler}>Agents</button> | <a href="#">Missions</a>
+                        </div>
+                    </nav>
```

Comment:
Since all of this is repeated in every part of your project, you could keep it outside of the conditional so you only have to write it once.